### ISSUE TITLE
GWAS: attempt to set 'colnames' on an object with less than two dimensions

### ISSUE THREAD

**Issue by naglemi** [QuantGen/BGData#51]
I'm attempting to run GWAS using BGData. I first went through the tutorial in the readme with test data and it worked properly. I'm now trying to run my data and am getting the following error.

```
> gwas <- GWAS(formula = my_phenotype ~ 1, data = bg)
Error in `colnames<-`(`*tmp*`, value = colnames(data@geno)[j]) : 
  attempt to set 'colnames' on an object with less than two dimensions
In addition: Warning message:
In parallel::mclapply(X = seq_len(nChunks), FUN = chunkApply, ...,  :
  all scheduled cores encountered errors in user code
```

The only thing I did differently from the tutorial was that I loaded the entire genome as a single .bed, rather than loading each chromosome separately and then combining them using ColumnLinkedMatrix. I don't know a  good way to load and combine chromosomes/scaffolds in my case since there are over 1000 for the reference genome I'm working with. I ran the following:

```
wg <- BEDMatrix("my_filepath.bed")
bg <- as.BGData(wg, alternatePhenotypeFile = "my_phenodata")
```

...and I can't see what the problem with the object "geno" is. It appears to have two dimensions, contrary to the error I'm getting from the GWAS function.
```
> bg
An object of class "BGData"
Slot "geno":
BEDMatrix: 882 x 4428726 [my_filepath.bed]
```

Thanks for the help getting this to run with my data.

**Comment by agrueneberg** [comment]
Thanks for the detailed report. Could you please confirm that the alternate phenotype file has been correctly loaded, i.e., if `bg@pheno` contains the `my_phenodata` column with valid data?

**Comment by naglemi** [comment]
It appears to me that it has been loaded correctly.

```
> bg@pheno
                 FID     IID my_phenotype
201782_400194 201782  400194   NA
201782_400495 201782  400495    0
```

Here's the tab-separated file I loaded in:
```
>head -n 3 REP_2_SW3_TDZ_withheader
FID	IID	my_phenotype
201782	400194	NA
201782	400495	0
```

**Comment by agrueneberg** [comment]
What version of BGData are you using? You can check with `packageVersion("BGData")`.

**Comment by naglemi** [comment]
I'm on version 2.0.0. Please let me know if anything else would be helpful to figure this out.

```
> packageVersion("BGData")
[1] ‘2.0.0’
```

**Comment by agrueneberg** [comment]
Sorry, I forgot to follow up on this issue... If you are still interested, could you please run your GWAS call again with `nCores = 1`, i.e. `gwas <- GWAS(formula = my_phenotype ~ 1, data = bg, nCores = 1)`? This should give you a better error message. The package we use for multi-core computations swallows error messages unless explicitly handled, which I have just addressed in 47e89bb40f7b20b6ddce4d1dc45d085ff93a8f86 (an alternative would be to try running the code again with the development version).
